source/MaterialXTest/MaterialXRenderOsl/GenReference.cpp

```diff
@@ -70,20 +70,18 @@ TEST_CASE("GenReference: OSL Reference", "[genreference]")
     bool failedGeneration = false;
     for (const mx::NodeDefPtr& nodedef : datalib->getNodeDefs())
     {
-        std::string nodeName = nodedef->getName();
-        std::string nodeNode = nodedef->getNodeString();
-        if (nodeName.size() > 3 && nodeName.substr(0, 3) == "ND_")
-        {
-            nodeName = nodeName.substr(3);
-        }
-
+        // determine the corresponding node for the nodedef
+        std::string nodeName = nodedef->getQualifiedName(nodedef->getNodeString());
```

Comment:
This looks much better to me, @ashwinbhat!

Just as a minor style note, we should make sure your newly added comments start with a capital letter, so that they match the surrounding style of the codebase.